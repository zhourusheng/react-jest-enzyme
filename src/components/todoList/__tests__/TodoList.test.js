import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, render, mount } from 'enzyme'
import TodoList from '..'


// setup设置
const props = {
  list: ['first', 'secord'],
  deleteTodo: jest.fn()
}

const setup = () => {
  const wrapper = shallow(<TodoList {...props} />)
  return {
    props,
    wrapper
  }
}

const setupByRender = () => {
  const wrapper = render(<TodoList {...props} />)
  return {
    props,
    wrapper
  }
}

const setupByMount = () => {
  const wrapper = mount(<TodoList {...props} />)
  return {
    props,
    wrapper
  }
}

// 使用 snapshot 进行UI测试
it('render current', () => {
  const tree = renderer.create(<TodoList {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
});


// 对组件节点进行测试
it('should has Button', () => {
  const { wrapper } = setup()
  expect(wrapper.find('Button').length).toBe(2)
});

it('should render 2 item', () => {
  const { wrapper } = setupByRender()
  expect(wrapper.find('button').length).toBe(2)
});

it('should render item equal', () => {
  const { wrapper } = setupByMount()
  wrapper.find('.item-text').forEach((node, index) => {
    expect(node.text()).toBe(wrapper.props().list[index])
  })
});

it('click item to be done', () => {
  const { wrapper } = setupByMount()
  wrapper.find('Button').at(0).simulate('click')
  expect(props.deleteTodo).toBeCalled()
});