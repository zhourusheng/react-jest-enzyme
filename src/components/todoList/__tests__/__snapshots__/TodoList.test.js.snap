// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render current 1`] = `
<div
  className="todo-list"
>
  <div>
    <span
      className="item-text"
    >
      first
    </span>
    <button
      className="ant-btn"
      onClick={[Function]}
      type="button"
    />
  </div>
  <div>
    <span
      className="item-text"
    >
      secord
    </span>
    <button
      className="ant-btn"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
